packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -352,12 +351,19 @@ const createHistoryService = ({ strapi }: { strapi: Core.Strapi }) => {
                   return currentRelationData;
                 }
 
+                const permissionChecker = getContentManagerService('permission-checker').create({
+                  userAbility: params.state.userAbility,
+                  model: 'plugin::upload.file',
+                });
+
                 const relatedEntry = await strapi.db
                   .query('plugin::upload.file')
                   .findOne({ where: { id: entry.id } });
 
-                if (relatedEntry) {
-                  currentRelationData.results.push(relatedEntry);
+                const sanitizedEntry = await permissionChecker.sanitizeOutput(relatedEntry);
```

Comment:
I wonder why you picked this in the end and not mainfield only population

Reply:
Here for upload I don't think a mainField is available 🤔 but maybe it is. For the other instance with relations I did test it out but I think I abandoned it quite quickly when values weren't showing up as expected. I'll try it again real quick.

Reply:
as you want I'm just curious, don't want to block you. And I doubt sanitization is needed on media assets

Reply:
Sounds good, I might just merge this now then and circle back to the idea in another PR with some of the refactor ideas.